NDTensors/src/Unwrap/test/runtests.jl

```diff
@@ -125,17 +128,57 @@ include("../../../test/device_list.jl")
   y = dev(randn(elt, 16))
   x = reshape(dev(randn(elt, 4, 4))', 16)
   copyto!(expose(y), expose(x))
-  @test y == x
-  @test copy(x) == x
+  @allowscalar begin
+    @test y == x
+    @test copy(x) == x
+  end
 
   y = dev(randn(elt, 8))
   x = @view reshape(dev(randn(elt, 8, 8))', 64)[1:8]
-  copyto!(expose(y), expose(x))
-  @test y == x
-  @test copy(x) == x
+  ## TODO This line fails on GPU with scalar indexing
+  @allowscalar begin
+    copyto!(expose(y), expose(x))
+    @test y == x
+    @test copy(x) == x
+  end
 
   y = Base.ReshapedArray(dev(randn(elt, 16)), (4, 4), ())
   x = dev(randn(elt, 4, 4))
   permutedims!(expose(y), expose(x), (2, 1))
   @test NDTensors.cpu(y) == transpose(NDTensors.cpu(x))
+
+  ##########################################
+  ### Testing an issue with CUDA&Metal transpose/adjoint mul
+  A = dev(randn(elt, (3, 2)))
+  B = dev(randn(elt, (3, 4)))
+  C = dev(randn(elt, (4, 2)))
+  Cp = copy(C)
+
+  ## This fails with scalar indexing
+  if dev != NDTensors.cpu
+    @test_broken mul!(transpose(C), transpose(A), B, true, false)
+  else
+    mul!(transpose(C), transpose(A), B, true, false)
```

Comment:
This is a bit funny to have here, it's not really testing anything and also overwrites `C`. Maybe we can just remove this branch.

Reply:
Yeah I wasn't sure if we needed to "check" the cpu version but I deleted the else branch from this and the Adjoint version